src/core/environment.go

```diff
@@ -561,19 +561,64 @@ func diskPartitions() (partitions []*proto.DiskPartition) {
 }
 
 func releaseInfo() (release *proto.ReleaseInfo) {
-	hostInfo, err := host.Info()
+	const osReleaseFile =  "/etc/os-release"
+	osRelease, err := getOsRelease(osReleaseFile)
 	if err != nil {
-		log.Errorf("Could not read release information for host: %v", err)
-		return &proto.ReleaseInfo{}
+		hostInfo, err := host.Info()
+		if err != nil {
+			log.Errorf("Could not read release information for host: %v", err)
+			return &proto.ReleaseInfo{}
+		}
+
+		return &proto.ReleaseInfo{
+			VersionId: hostInfo.PlatformVersion,
+			Version:   hostInfo.KernelVersion,
+			Codename:  hostInfo.OS,
+			Name:      hostInfo.PlatformFamily,
+			Id:        hostInfo.Platform,
+		}
 	}
+	return osRelease
+}
 
-	return &proto.ReleaseInfo{
-		VersionId: hostInfo.PlatformVersion,
-		Version:   hostInfo.KernelVersion,
-		Codename:  hostInfo.OS,
-		Name:      hostInfo.PlatformFamily,
-		Id:        hostInfo.Platform,
+// getOsRelease reads osReleaseFile and returns release information for host.
+// If os.Stat(osReleaseFilePath) does not find file, or
+// ioutil.ReadFile(osReleaseFilePath) fails to read file, an error occurs.
+func getOsRelease(osReleaseFile string) (release *proto.ReleaseInfo, err error) {
+	_ , osReleaseError := os.Stat(osReleaseFile)
+	if os.IsNotExist(osReleaseError) {
+		log.Errorf("Could not find path for os-release file on the host")
+		return &proto.ReleaseInfo{}, errors.New("unable to find " + osReleaseFile)
+	}
```

Comment:
a couple things with this error handling

1. `os.IsNotExist` is deprecated, they recommend using `errors.Is(err, fs.ErrNotExist)`
2. when our function returns multiple values with the error, it's more common to return zero-values for everything except the error, in this case `nil` instead of `&proto.ReleaseInfo{}`; we don't need to allocate some memory to point to an empty struct
3. `ErrNotExist` is not the only error we could get from `Stat`; this code is ignoring all other errors and continuing on as if `Stat` succeeded. If we haven't handled the error, we want to return it to the caller.
4. a ramble on logging and golang error handling:

Unlike languages with exceptions, we deal with golang errors at every level of the call stack. A common pattern in other languages is to log a problem, then throw an exception that will be handled _presumably_ by some distant piece of code. With go, we know the caller is going to have to deal with the error, even if that's just passing it along to the next caller.

If we use the "log + return error" strategy at each level of the call stack, we're going to get a lot of log messages for one error, with each log message containing different details. This can get noisy and hard to decipher. Additionally, those details can be useful for error handling, but they are trapped in the log messages.

In this specific case, the caller has no way to tell that the root error is a "file not found"; it can't make different choices based on the returned error.

Golang gives us an solution to this with [error wrapping](https://go.dev/blog/go1.13-errors), letting us add context to an error before passing it back to the caller.

```suggestion
	if osReleaseError != nil {
		return nil, fmt.Errorf("release file %s unavailable: %w", osReleaseFile, osReleaseError)
	}
```

In our case, the caller (`releaseInfo`) wants to handle this error by falling back to `host.Info()`. It could log the error in a `log.Warnf` and then proceed with the fallback.

Reply:
This was an interesting `Go` take on `error`. I will have this updated.